src/storage/posix/storage.c

```diff
@@ -452,28 +463,39 @@ storagePosixPathRemove(THIS_VOID, const String *path, bool recurse, StorageInter
 
     MEM_CONTEXT_TEMP_BEGIN()
     {
-        // Recurse if requested
-        if (recurse)
-        {
-            StoragePosixPathRemoveData data =
-            {
-                .driver = this,
-                .path = path,
-            };
+        bool removed = false;
 
-            // Remove all sub paths/files
-            storageInterfaceInfoListP(this, path, storageInfoLevelExists, storagePosixPathRemoveCallback, &data);
-        }
-
-        // Delete the path
-        if (rmdir(strZ(path)) == -1)
+        do
```

Comment:
This would be a great place to leave a comment 1) keep trying until the directory is fully removed, and 2) why it might not work the first time.

Can we guarantee it can't go into an infinite loop?  Perhaps a limit on how many times we retry?

Reply:
Comment updated in 8fcb15cf4f8 and b36d5afce.

> Can we guarantee it can't go into an infinite loop? Perhaps a limit on how many times we retry?

It seems unlikely, but not impossible. A retry limit would be a good here -- that's pretty standard practice elsewhere. That requires shimming something, but not that big a deal.